### PR TITLE
ci: run full ui package tests

### LINKED ISSUE #869: [Bug] 34 tests under packages/ui/src/** are silently excluded from CI
### What happened?

34 `*.test.ts` / `*.test.tsx` files under `packages/ui/src/**` are not executed by any CI job. Any contract drift in those tests goes undetected.

### Which area seems affected?

CI, release, or developer tooling

### How much does this affect you?

Breaks an important workflow

### Steps to reproduce

1. Push a PR with changes to any file under `packages/ui/src/`
2. CI completes all jobs green
3. The corresponding `*.test.ts` files are never run — no test output visible in any CI log

### What did you expect to happen?

Tests under `packages/ui/src/` should execute as part of CI and fail the build when they break.

### Evidence

`.github/workflows/ci.yml` has three unit entry points:

- `unit-app`: `bun turbo test:ci --filter=@opencode-ai/app`
- `unit-opencode`: `bun turbo test:ci --filter=opencode`
- `unit-desktop`: `bun turbo test:ci --filter=@opencode-ai/desktop-electron`

`turbo.json` does **not** define a `@opencode-ai/ui#test:ci` task, so turbo's dependency graph never propagates into ui tests.

The `unit-ui-focused` job only runs 5 token/theme/typography guard tests under `packages/ui/test/`. The 32 tests under `packages/ui/src/components/` (markdown, scroll-view, picker, tool-contract, session-turn-*, etc.) plus 2 under `packages/ui/src/util/`, `lib/`, `i18n/` are all silent.

### Trigger

PR #861 (Slice 3 extracting TodoStatusMarker) added `packages/ui/src/components/todo-status-marker.test.ts`. The external reviewer noticed it falls into this black hole.

### Diagnostics

See [CI workflow](https://github.com/Astro-Han/pawwork/blob/dev/.github/workflows/ci.yml) and `turbo.json` for the current configuration.

### Fix directions (to be evaluated)

A. **Expand `unit-ui-focused` listing** — explicitly list all `src/*.test.ts` files. Downside: every new test requires a ci.yml edit, and the labeler.yml auto-adds `task` label.

B. **Add `test:ci` script in `packages/ui/package.json` and define `@opencode-ai/ui#test:ci` in turbo.json** — aligns with how app/opencode/desktop do it, zero ci.yml changes, future tests auto-included. **Recommended**.

C. **Run `bun --cwd packages/ui test src/` in `unit-ui-focused`** — downside: mixes with the fast-guard purpose and may slow CI.

### Acceptance

- Any PR editing `packages/ui/src/**` shows corresponding `*.test.ts` execution in CI logs
- Intentionally breaking a test under `packages/ui/src/` causes CI to fail

### PR DESCRIPTION
**Pull request by Astro-Han**
## Summary

Run the full `packages/ui` package test suite in CI instead of only the focused tests under `packages/ui/test`.

## Why

`packages/ui/src/**` already contains package-level tests, but CI did not invoke a UI package `test:ci` task. That left 34 source-adjacent tests silent on pull requests.

## Related Issue

Closes #869

## Human Review Status

Pending

## Review Focus

Please focus on whether the new `unit-ui` job and Turbo `@opencode-ai/ui#test:ci` task cover the intended UI package tests without widening CI scope beyond the package boundary.

## Risk Notes

Low risk: this changes CI coverage and aligns two UI control assertions with the current design tokens. No runtime dependency or platform packaging behavior changed.

Skipped conditional checklist items:

- Platform impact: no platform, packaging, updater, signing, path, shell, or permissions surface was touched.
- Docs/release/dependency/local artifact impact: no docs, release notes, dependencies, credentials, deletion behavior, generated content, or local artifacts were committed.

## How To Verify

```text
bun --cwd packages/opencode test test/github/ci-workflow.test.ts
Result: 11 passed, 0 failed.

bun turbo test:ci --filter=@opencode-ai/ui --dry=json | jq -r '.tasks[].taskId'
Result: includes @opencode-ai/ui#test:ci, with build dependencies only.

bun turbo test:ci --filter=@opencode-ai/ui
Result: 648 passed, 0 failed across 51 files.

bun run snap sidebar
Result: 1 passed; reviewed the generated sidebar preview locally.
```

## Screenshots or Recordings

Visual check performed with `bun run snap sidebar`; reviewed `docs/design/preview/screenshots/sidebar.png` locally for the changed sidebar/titlebar control styling.

## Checklist

> **How to use this checklist:**
>
> - Tick a box by replacing `[ ]` with `[x]`. Do not edit, add, or remove items.
> - The bot-applied label items can only be honestly ticked AFTER the PR is opened and the labeler / priority-triage bots have run — return to the PR description and tick them then.
> - Most items are required. The few that are conditional are explicitly marked **(conditional)**; for those, leave unticked if they truly do not apply and explain why in Risk Notes. All other items must be ticked before requesting human review.

- [x] **Type label** — this PR carries exactly one of `bug`, `enhancement`, `task`, `documentation`. Type labels are author-added; the labeler bot does NOT assign them. Add the label in the GitHub UI, then tick this.
- [x] **Routing labels** — this PR carries at least one of `app`, `ui`, `platform`, `harness`, `ci`. The labeler bot assigns these on PR open based on changed paths. Confirm the bot's choice (or override if wrong), then tick this.
- [x] **Priority label** — this PR carries exactly one of `P0`, `P1`, `P2`, `P3`. The priority-triage bot suggests one on PR open. Confirm or override, then tick this.
- [x] Human Review Status above is set to `Pending`, `Approved by @<reviewer>`, or `Not required: <reason>` (default is `Pending`; "not required" is restricted to bot-authored low-risk PRs).
- [x] I linked the related issue, or stated in Summary why there is no issue.
- [x] I described the review focus and any meaningful risks.
- [x] I replaced the example block in How To Verify with the real verification steps and the key result for each.
- [x] I did not introduce unrelated refactors, dependencies, generated files, or file changes beyond the stated scope.
- [x] **(conditional)** I manually checked visible UI or copy changes when needed, with screenshots or recordings. Leave unticked only if no visible UI or copy changed.
- [ ] **(conditional)** I considered macOS and Windows impact for platform, packaging, updater, signing, paths, shell, or permissions changes. Leave unticked only if no platform/packaging surface was touched.
- [ ] **(conditional)** I called out docs, release notes, dependencies, permissions, credentials, deletion behavior, generated content, or local file changes when relevant. Leave unticked only if none of those surfaces was touched.
- [x] I reviewed the final diff for unrelated changes and suspicious dependency changes.
- [x] I am targeting `dev`, and my PR title and commit messages use Conventional Commits in English.

### CODE REVIEW

File: packages/opencode/test/github/ci-workflow.test.ts [hunk 1]
```diff
@@ -154,6 +161,21 @@ function isWindowsOpencodeShard(item: Record<string, unknown>): item is { comman
...
+    }
+    const uiPackage = JSON.parse(readFileSync(uiPackageJsonPath, "utf8")) as {
+      scripts?: Record<string, string>
+    }
+
+    expect(uiPackage.scripts?.["test:ci"]).toBe(
+      "mkdir -p .artifacts/unit && bun test --reporter=junit --reporter-outfile=.artifacts/unit/junit.xml",
+    )
+    expect(turbo.tasks?.["@opencode-ai/ui#test:ci"]?.outputs).toEqual([".artifacts/unit/junit.xml"])
+    expect(turbo.tasks?.["@opencode-ai/ui#test:ci"]?.passThroughEnv).toEqual(["*"])
```
[the remark this answers is not in the document]
Comment (Astro-Han) [reply packages/opencode/test/github/ci-workflow.test.ts:177]: Fixed in 1810f4c924 by asserting `@opencode-ai/ui#test:ci` keeps `dependsOn: ["^build"]`. Verified with `bun --cwd packages/opencode test test/github/ci-workflow.test.ts` and the latest PR checks show `unit-opencode`, `unit-ui`, and `ci / check` passing.
